src/client/metric/countvector.rs

```diff
@@ -0,0 +1,149 @@
+use super::*;
+
+/// A count vector for multiple strictly increasing integer values, in possibly
+/// varying increments
+///
+/// Internally uses an `InstanceMetric<u64>` with `Semantics::Counter` and
+/// `Count::One` scale, and `1` count dimension
+pub struct CountVector {
+    im: InstanceMetric<u64>,
+    indom: Indom,
+    init_val: u64
+}
+
+impl CountVector {
+    /// Creates a new count vector with given initial value and instances
+    pub fn new(name: &str, init_val: u64, instances: &[&str],
```

Comment:
Although since this simplistic mode of initialization does work for GaugeVector we should , probably keep it, at the risk of different APIs for CounterVector and GaugeVector, which might lead to frequent confusion and mistakes